### PR TITLE
Mira 4292 gql 16 update

### PR DESCRIPTION
**Pull request by sgohlke**
Migrate logic to support graphql-js version 16.x.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,7 +31,7 @@ TypeScript declarations are provided within the project.
 - Uses out-of-the-box default options to ease use and keep code short
 - Provides hot reloading for schema and options
 - Provides out-of-the-box metrics for GraphQLServer
-- Uses and is compatible with `graphql-js` library version 14 and 15.
+- Uses and is compatible with `graphql-js` library version 15 and 16.
```
Comment (jenssuhr) [inline README.md:34]: Is this version still compatible with `graphql-js` 15?

File: tests/logger/LogHelper.test.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 /* eslint-disable max-len */
-import {GraphQLError} from 'graphql'
+import {GraphQLError,
+    Kind} from 'graphql'
```
Comment (jenssuhr) [inline tests/logger/LogHelper.test.ts:3]: I think we should update the formatter, this looks really weird to me.

File: tests/logger/LogHelper.test.ts [hunk 3]
```diff
@@ -32,22 +33,18 @@ const graphQLErrorWithVariables: GraphQLError = new GraphQLError(messageWithVari
...
-        undefined,
-        undefined,
-        undefined,
-        undefined
-        , {exception: 'A stacktrace', serviceName: 'customer'})
+const graphQLErrorWithSourceBody = new GraphQLError('`CustomerPayload` is an extension type',{
+    source: {get [Symbol.toStringTag](): string {
+        return this[Symbol.toStringTag]
+    }, locationOffset: {line: 1, column: 1}, name: 'doesnotmatter', body: customerQuery},
+    extensions: {exception: 'A stacktrace', serviceName: 'customer'}
```
Comment (jenssuhr) [inline tests/logger/LogHelper.test.ts:40]: Can you add some line breaks to this constructor? It's a bit hard to grasp which brackets belong together.

**Review by jenssuhr — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (sgohlke) [reply README.md:34]: The application v 1.x is compatible with graphql-js 15 while v 2.x is compatible with graphql-js 16.

File: tests/logger/LogHelper.test.ts (hunk 2 above)
Comment (sgohlke) [reply tests/logger/LogHelper.test.ts:3]: What's your proposal on how to format this line?

File: tests/logger/LogHelper.test.ts (hunk 3 above)
Comment (sgohlke) [reply tests/logger/LogHelper.test.ts:47]: Done

File: tests/logger/LogHelper.test.ts (hunk 2 above)
Comment (sgohlke) [reply tests/logger/LogHelper.test.ts]: I reformated it with multiline = true option. Does it look better now?

**Review by pro-ship-it — COMMENTED** [review]
LGTM

File: README.md (hunk 1 above)
Comment (jenssuhr) [reply README.md:34]: OK, would it make sense to add that here, even though we have the table above this text?

File: tests/logger/LogHelper.test.ts (hunk 2 above)
Comment (jenssuhr) [reply tests/logger/LogHelper.test.ts]: Yes, much better 👍

File: README.md (hunk 1 above)
Comment (sgohlke) [reply README.md]: Added the info

**Review by jenssuhr — APPROVED** [review]

**Pull request merged**